ops/model.py

```diff
@@ -262,6 +262,24 @@ def status(self, value: 'StatusBase'):
         self._backend.status_set(value.name, value.message, is_app=True)
         self._status = value
 
+    def planned_units(self) -> int:
+        """Get the number of units that Juju has "planned" for this application.
+
+        E.g., if an operator ran "juju deploy foo", then "juju add-unit -n 2 foo", the
+        planned unit count for foo would be 3.
+
+        We deliberately do not attempt to inspect whether these units are actually running
+        or not. That is a task left up to the future, when goal state is more mature.
```

Comment:
It'd be nice to not refer to "goal state" here. We want to kill that idea altogether in juju, in the sense that the number of pending units is just part of the current state like everything else.